var/spack/repos/builtin/packages/dakota/package.py

```diff
@@ -44,6 +44,7 @@ class Dakota(CMakePackage):
     depends_on('mpi', when='+mpi')
 
     depends_on('python')
+    depends_on('perl-data-dumper', type='build', when='@:6.12')
```

Comment:
```suggestion
    depends_on('perl-data-dumper', type='build', when='@:6.12')
    depends_on('perl-data-dumper', type='build', when='@6.12:')
```

`@6.12:` is 6.12 and above, `@:6.12` is everything strictly < 6.13